Client/game_sa/CAutomobileSA.h

```diff
@@ -163,19 +159,9 @@ static_assert(sizeof(CAutomobileSAInterface) == 0x988, "Invalid size for CAutomo
 
 class CAutomobileSA : public virtual CAutomobile, public virtual CVehicleSA
 {
-private:
-    CDoorSA* door[MAX_DOORS];
-
 public:
-    CAutomobileSA(eVehicleTypes dwModelID, unsigned char ucVariation, unsigned char ucVariation2);
-    CAutomobileSA(CAutomobileSAInterface* automobile);
-    ~CAutomobileSA();
-
-    bool BurstTyre(DWORD dwTyreID);
-    bool BreakTowLink();
-
-    bool       IsComponentPresent(int iComponentID);
-    CPhysical* SpawnFlyingComponent(int iCarNodeIndex, int iUnknown);
+    CAutomobileSA() = default;
```

Comment:
You didn't define a default constructor for the other classes, why did you do it here?

Reply:
CAutomobileSA will not have an implicitly-defined constructor because we have already one defined which takes a CAutomobileSAInterface*.
We need a default constructor for inherited classes to call because we don't pass the vehicle pointer down.
CHeliSA(CHeliSAInterface*) will automatically call CAutomobileSA(), CVehicleSA(), CPhysicalSA() and CEntitySA().